### PR TITLE
Windows Installer: Support non-admin installs.

### PR DESCRIPTION
**Pull request by mook-as**
This adds support for non-admin installs.  This is done via Windows Installer UI where one page of the wizard lets the user switch between for all users (privileged) or for the current user only (unprivileged).

This also moves most of the XML into a mustache template for easier editing (since it's now more complicated).

Part of #3309.

### CODE REVIEW

File: scripts/lib/installer-win32-gen.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+const ComponentGroup = 'ComponentGroup';
+const ComponentGroupRef = 'ComponentGroupRef';
+const Directory = 'Directory';
+const File = 'File';
+const Fragment = 'Fragment';
+const Shortcut = 'Shortcut';
+const ShortcutProperty = 'ShortcutProperty';
+
+/**
+ * A structure representing the files (but not subdirectories) within a
```
Comment (Nino-K) [inline scripts/lib/installer-win32-gen.tsx:80]: I find the comment a little confusing, it says `but not subdirectories` however, I see a directories array (line: 89) within the directory type, isn't that meant to represent sub-dirs?

File: scripts/lib/installer-win32-gen.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      }
+    }));
+
+    result.directories.sort((a, b) => a.name.localeCompare(b.name));
+
+    files.sort((a, b) => a.name.localeCompare(b.name));
+    result.files = files;
+
+    files.forEach(f => hasher.update(f.id));
+    result.id = `d_${ hasher.digest('base64url').replaceAll('-', '.') }`;
```
Comment (Nino-K) [inline scripts/lib/installer-win32-gen.tsx:129]: Per our offline conversation with Mark, should this be consistent with the id on line 115`?

**Review by Nino-K — CHANGES_REQUESTED** [review]

**Comment by Nino-K** [comment]
Just a few things that require @mook-as confirming.

File: scripts/lib/installer-win32-gen.tsx (hunk 2 above)
Comment (mook-as) [reply scripts/lib/installer-win32-gen.tsx:128]: Yep, added the underscore (`f_`) on line 115 to match.

File: scripts/lib/installer-win32-gen.tsx (hunk 1 above)
Comment (mook-as) [reply scripts/lib/installer-win32-gen.tsx]: Right, the comment was out of date; updated (to include subdirectories).

**Review by Nino-K — APPROVED** [review]
LGTM

**Comment by wiruzman** [comment]
I know this is merged, but I cannot control `ALLUSERS` and `MSIINSTALLPERUSER` from the console, would it be possible?

**Comment by mook-as** [comment]
Hi @wiruzman! it would work better to make a discussion topic (or an new issue, depending on if you can identify a code change that would make things better).  Please do not resurrect old threads  (much like old-school forums). Thanks!

**Pull request merged**